### PR TITLE
Layout

### PR DESCRIPTION
**Pull request by CorneliaKelinske**
@alanvardy quick code review please
I have implemented both authentication and authorization now, including the corresponding tests.
I also have a delete function now (available only to the Admin).

### CODE REVIEW

File: lib/the_little_thinkers_space/accounts/user.ex [hunk 1]
```diff
@@ -52,6 +52,11 @@ defmodule TheLittleThinkersSpace.Accounts.User do
   end
 
   def authorize(_, %User{role: "Admin"}, _user), do: :ok
-  def authorize(:index, %User{}, _user), do: :ok
+
+  def authorize(action, %User{role: "The Little Thinker"}, _user) when action in [:index, :show],
+    do: :ok
```
Comment (alanvardy) [inline lib/the_little_thinkers_space/accounts/user.ex:57]: Make this a multi-line function with do and end?

File: lib/the_little_thinkers_space_web/controllers/fallback_controller.ex [hunk 2]
```diff
@@ -4,6 +4,6 @@ defmodule TheLittleThinkersSpaceWeb.FallbackController do
   def call(conn, {:error, :unauthorized}) do
     conn
     |> put_flash(:error, "You are not allowed to do this!")
-    |> redirect(to: Routes.upload_path(conn, :index))
+    |> redirect(to: Routes.page_path(conn, :home))
```
Comment (alanvardy) [inline lib/the_little_thinkers_space_web/controllers/fallback_controller.ex:7]: :+1:

File: lib/the_little_thinkers_space_web/controllers/user_controller.ex [hunk 3]
```diff
@@ -38,10 +41,28 @@ defmodule TheLittleThinkersSpaceWeb.UserController do
...
   end
 
-  # def show(conn, %{"id" => id}) do
-  #   user = Accounts.get_user!(id)
-  #   render(conn, "show.html", user: user)
-  # end
+  def show(conn, %{"id" => id}) do
+    current_user = conn.assigns.current_user
+
+    user = Accounts.get_user!(id)
```
Comment (alanvardy) [inline lib/the_little_thinkers_space_web/controllers/user_controller.ex:47]: If the user is not found, does this result in a 404 or a server error?

Comment (alanvardy) [reply lib/the_little_thinkers_space_web/controllers/user_controller.ex:47]: You can test by going to /users/99 or whatever

File: lib/the_little_thinkers_space_web/templates/layout/app.html.eex [hunk 4]
```diff
@@ -11,21 +11,11 @@
...
-            <li><%= link "Register", to: Routes.user_path(@conn, :new) %></li>
-            <li><%= link "Log in", to: Routes.session_path(@conn, :new) %></li>
-           <% end %>
-          </ul>
-        </nav>
+       <%= if @current_user do %>
+       <%= render "navigation_home.html", conn: @conn, current_user: @current_user %>
+      <% else %>
+       <%= render "navigation_landing.html", conn: @conn %>
+      <% end %>
```
Comment (alanvardy) [inline lib/the_little_thinkers_space_web/templates/layout/app.html.eex:18]: Indentation seems a bit wonky. There might be some extensions that can help you with formatting.

File: lib/the_little_thinkers_space_web/templates/layout/navigation_home.html.eex [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+            <% end %>
+
+            <%= if @current_user.role == "Admin" do %>
+            <li><%= link "Register new user", to: Routes.user_path(@conn, :new) %></li>
+             <% else %>
+            <li> Contact </li>
+            <% end %>
+
+
+
```
Comment (alanvardy) [inline lib/the_little_thinkers_space_web/templates/layout/navigation_home.html.eex:16]: Random gaps

File: lib/the_little_thinkers_space_web/templates/layout/navigation_home.html.eex [hunk 6]
```diff
@@ -0,0 +1,23 @@
+          <nav role="navigation">
+          <ul>
+            <%= if @current_user.role in ["Admin", "The Little Thinker"] do %>
+            <li><%= link "Ulrik's Crew", to: Routes.user_path(@conn, :index) %></li>
+             <% else %>
+            <li><%= link "#{@current_user.name}", to: Routes.user_path(@conn, :show, @current_user.id)%>
+            <% end %>
```
Comment (alanvardy) [inline lib/the_little_thinkers_space_web/templates/layout/navigation_home.html.eex:7]: Indentation

File: lib/the_little_thinkers_space_web/templates/layout/navigation_landing.html.eex [hunk 7]
```diff
@@ -0,0 +1,8 @@
+
+
```
Comment (alanvardy) [inline lib/the_little_thinkers_space_web/templates/layout/navigation_landing.html.eex:2]: gaps

File: test/the_little_thinkers_space_web/controllers/user_controller_test.exs [hunk 8]
```diff
@@ -74,13 +85,57 @@ defmodule TheLittleThinkersSpaceWeb.UserControllerTest do
...
       conn = post(conn, Routes.user_path(conn, :create), user: UserHelper.create_attrs())
-      assert html_response(conn, 302) =~ "<a href=\"/uploads\">redirected</a>."
+      assert html_response(conn, 302) =~ "<a href=\"/home\">redirected</a>."
+    end
+  end
+
+  describe "show user" do
+    setup [:create_admin_user]
+    setup [:create_user]
+    setup [:create_little_thinker]
```
Comment (alanvardy) [inline test/the_little_thinkers_space_web/controllers/user_controller_test.exs:101]: These can all be on one line

```suggestion
    setup [:create_admin_user, :create_user, :create_little_thinker]
```

**Review by alanvardy — APPROVED** [review]

File: lib/the_little_thinkers_space_web/controllers/user_controller.ex (hunk 3 above)
Comment (CorneliaKelinske) [reply lib/the_little_thinkers_space_web/controllers/user_controller.ex:47]: @alanvardy I have done a little something. but this requires another review. cf. later comment accompanying the re-review request.

File: lib/the_little_thinkers_space_web/templates/layout/app.html.eex (hunk 4 above)
Comment (CorneliaKelinske) [reply lib/the_little_thinkers_space_web/templates/layout/app.html.eex:18]: Thank you! will look into this

**Comment by CorneliaKelinske** [comment]
@alanvardy I implemented the "Not found" for the show_user(id) function. But I had to change my controller tests accordingly. It would be great if you could have a look at this. Also, I am getting warnings about deprecated code. Would love to address this with you.
On a different note: if I implement a "not found" for the user show function, shouldn't it be implemented for the upload show function as well?
thank you!

**Review by alanvardy — COMMENTED** [review]
LGTM :+1:

**Pull request merged**